testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -1154,6 +1154,38 @@ private void assertSelectsOnSpecialCharacters(String trinoTableName, String spar
         }
     }
 
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testTrinoReadsSparkSortOrder()
+    {
+        String sourceTableNameBase = "test_insert_into_sorted_table_" + randomNameSuffix();
+        String trinoTableName = trinoTableName(sourceTableNameBase);
+        String sparkTableName = sparkTableName(sourceTableNameBase);
+
+        onSpark().executeQuery("CREATE TABLE " + sparkTableName + " (a INT, b INT, c INT) USING ICEBERG ");
+        onSpark().executeQuery("ALTER TABLE " + sparkTableName + " WRITE ORDERED BY b, c DESC NULLS LAST");
+
+        Assertions.assertThat((String) onTrino().executeQuery("SHOW CREATE TABLE " + trinoTableName).getOnlyValue())
+                .contains("sorted_by = ARRAY['b ASC NULLS FIRST','c DESC NULLS LAST']");
+
+        onTrino().executeQuery("INSERT INTO " + trinoTableName + " VALUES (1, 2, 3), (3, 2, 1), (NULL, NULL, NULL)");
+    }
+
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testTrinoIgnoresUnsupportedSparkSortOrder()
+    {
+        String sourceTableNameBase = "test_insert_into_sorted_table_" + randomNameSuffix();
+        String trinoTableName = trinoTableName(sourceTableNameBase);
+        String sparkTableName = sparkTableName(sourceTableNameBase);
+
+        onSpark().executeQuery("CREATE TABLE " + sparkTableName + " (a INT, b INT, c INT) USING ICEBERG ");
+        onSpark().executeQuery("ALTER TABLE " + sparkTableName + " WRITE ORDERED BY truncate(b, 3)");
+
+        Assertions.assertThat((String) onTrino().executeQuery("SHOW CREATE TABLE " + trinoTableName).getOnlyValue())
+                .doesNotContain("sorted_by");
+
+        onTrino().executeQuery("INSERT INTO " + trinoTableName + " VALUES (1, 2, 3), (3, 2, 1), (NULL, NULL, NULL)");
```

Comment:
From https://iceberg.apache.org/spec/#sorting

> A data or delete file is associated with a sort order by the sort order’s id within [a manifest](https://iceberg.apache.org/spec/#manifests). Therefore, the table must declare all the sort orders for lookup.

Could you please inspect the metadata file written by Trino to ensure that it doesn't have any `sort-orders` ? 
I think a combination of:
-  retrieving the metadata file (see   https://github.com/trinodb/trino/blob/390c094273d20f54749fee737b9dbd4377e906e0/testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java#L2431-L2433 )
- retrieving the metadata file in memory
```
ByteArrayOutputStream bos = new ByteArrayOutputStream();
hdfsClient.loadFile(source + "/" + file, bos);
```
-  construct the table metadata `org.apache.iceberg.TableMetadataParser#fromJson(java.lang.String)` 
 
may help to do this.

Reply:
Looked a little closer at this, I don't think that field is set on the manifest even if we do sort the file. I'm missing a line here:  https://github.com/trinodb/trino/blob/master/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java#L806
```.withSortOrder(sortOrder)```

I wonder what this field is actually used for, though. Maybe Spark avoids sorting data if it's already sorted?

Reply:
> I wonder what this field is actually used for, though. Maybe Spark avoids sorting data if it's already sorted?

like https://github.com/trinodb/trino/pull/6634?

>  I don't think that field is set on the manifest even if we do sort the file. 

we definitely need to be storing the sorting information somewhere, otherwise how would the future writes know they should sort?

Reply:
> we definitely need to be storing the sorting information somewhere, otherwise how would the future writes know they should sort?

We do set the `default-sort-order-id` which tells future writes which sort order they should use.

The field Marius is talking about is attached to the individual [Manifests](https://iceberg.apache.org/spec/#manifests). I think it is okay for us not to be setting this value for now.